python/tvm/script/tir/ty.py

```diff
@@ -46,6 +46,19 @@ def evaluate(self):
         return tvm.ir.PrimType(self.type)
 
 
+class ConcreteBufferType(TypeGeneric):  # pylint: disable=too-few-public-methods, abstract-method
+    """TVM script typing class for uniform Type objects"""
+
+    def __init__(self, vtype):
+        self.type = vtype
+
+    def evaluate(self):
+        return tvm.ir.PrimType(self.type)
+
+    def __call__(self, shape, dtype, elem_offset):
```

Comment:
let's also support other arguments of `decl_buffer`:
https://github.com/apache/tvm/blob/main/python/tvm/script/tir/special_stmt.py#L154-L159